pkg/storaged/client.js

```diff
@@ -411,6 +396,41 @@ function btrfs_start_polling() {
     btrfs_findmnt_poll();
 }
 
+/* Derived indices.
+ */
+
+function is_multipath_master(block) {
+    // The master has "mpath" in its device mapper UUID.  In the
+    // future, storaged will hopefully provide this information
+    // directly.
+    if (block.Symlinks && block.Symlinks.length) {
+        for (let i = 0; i < block.Symlinks.length; i++)
+            if (utils.decode_filename(block.Symlinks[i]).indexOf("/dev/disk/by-id/dm-uuid-mpath-") === 0)
+                return true;
+    }
+    return false;
+}
+
+function is_toplevel_drive(block) {
+    // We consider all Block objects that point to the same Drive
+    // objects to be multipath members for a single actual device.
+    //
+    // However, objects for partitions point to the same Drive object
+    // as the object for the partition table. We have to ignore them.
+
+    if (client.blocks_part[block.path])
+        return false;
+
+    // Also, eMMCs have special partition-like sub-devices that point
+    // to the main Drive. We identify them by their name, just like
+    // UDisks2.
+
+    if (utils.decode_filename(block.Device).match(/^mmcblk[0-9]boot[0-9]$/))
```

Comment:
This won't work, the devices have a `/dev/` prefix

Reply:
```
-    if (utils.decode_filename(block.Device).match(/^mmcblk[0-9]boot[0-9]$/))
+    if (utils.decode_filename(block.Device).match(/\/dev\/mmcblk[0-9]boot[0-9]$/))
```

This works.

Reply:
Also our `add_disk` code does this:

```
 559   │         if not serial:
 560   │             serial = f"DISK{index}"
 561   │         dev = 'sd' + string.ascii_lowercase[index]
 562   │         extra = "<boot order='1'/>" if boot_disk else ""
 563   │         disk_desc = TEST_DISK_XML % {
 564   │             'file': image,
 565   │             'serial': serial,
 566   │             'unit': index,
 567   │             'dev': dev,
 568   │             'type': image_type,
 569   │             'extra': extra,
 570   │         }
```

So the device name seems user controlled, we could add a test for that? Or maybe via the `ram_disk` route?

Reply:
> So the device name seems user controlled

Not really... I think the `dev = 'sd' + index` is just a good guess at what the device will be called in the guest. It doesn't control it.

Reply:
> This won't work, the devices have a `/dev/` prefix

Ouch, that's why test, I guess. :-) Thanks!